### DISCUSSION TITLE
Service to Save Persistent states

### DISCUSSION THREAD

**Discussion by r-t-s** [home-assistant/architecture#594] [General] [answered]
Currently persistent states (for entities derived from RestoreEntity) are saved at  hass startup, every 15 minutes there after, and prior to shutdown.

If the power goes down and the entity (i.e. maybe a device) has no way to save some state information internally, then any state changes sinse the last save are lost.

My actual use case is for a  timer used with an automation. It's imperitive to to stop an action that was started at a later time.
For power lost before the 15 minute timer  interval, the state information from this action is lost, and is not able to be recovered.

A service to force the state to be saved immediately and then resume the 15 minute interval would allow an automation to reduce the vulnerable interval to ms instead of up to 15 minutes.

I propose adding a service:  homeassistant.save_persistent_states

**Comment by MartinHjelmare** [comment, marked as answer]
This was solved here:
https://github.com/home-assistant/core/pull/53881
